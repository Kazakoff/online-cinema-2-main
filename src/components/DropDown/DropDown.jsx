import React, { useRef, useState, useEffect } from 'react';

export const DropDown = (props) => {

    const {
        label,
        children,
        style,
    } = props

    const [menuSize, setMenuSize] = useState('auto')
    const buttonRef = useRef(null);
    const menuRef = useRef(null);

    useEffect(() => {
        if (buttonRef.current && menuRef.current) {
            setMenuSize(handleCompareSize)
        }
    }, [buttonRef.current?.offsetWidth, menuRef.current?.offsetWidth])

    const handleCompareSize = () => {
        if (buttonRef.current.offsetWidth > menuRef.current.offsetWidth) {
            return buttonRef.current.offsetWidth
        } else {
            return 'auto'
        }
    }

    return (
        <div className={'btn-group dropdown'} style={style}>

            <button ref={buttonRef} type='button' className='btn whiteBtn dropdown-toggle'
                data-toggle='dropdown' aria-haspopup='true' aria-expanded='false'>{label}</button>


            <div ref={menuRef} className='dropdown-menu' style={{ minWidth: menuSize }} >
                <div style={{ height: children.length > 6 ? '155px' : 'auto', overflowY: 'auto', whiteSpace: 'nowrap', paddingRight: '5px', marginLeft: '5px' }}>
                    {children}
                </div>
            </div>

        </div>
    )
}