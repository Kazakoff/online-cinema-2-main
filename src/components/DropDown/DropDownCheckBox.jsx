import React from 'react';

export const DropDownCheckBox = (props) => {

    const {
        label,
        checked,
        onChange,
    } = props

    return (
        <div>
            <input
                type='checkbox'
                style={{ marginRight: '5px' }}
                checked={checked}
                onChange={onChange}
            />
            {label}<br />
        </div>
    )
}