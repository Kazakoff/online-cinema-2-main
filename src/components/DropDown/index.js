export { DropDownCheckBox } from './DropDownCheckBox'
export { DropDown } from './DropDown'