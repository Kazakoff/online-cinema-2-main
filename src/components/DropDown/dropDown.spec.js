import React from 'react'
import { DropDown } from './DropDown'

const props = {
    label: 'some label',
    style: { marginLeft: '7px' },
    children: 
    <>
        <p>1</p>
        <p>2</p>
    </>
};

const setUp = (props) => shallow(<DropDown {...props} />);

describe('DropDown component', () => {
    it('should render DropDown component', () => {
        const component = setUp(props)
        expect(component.debug()).toMatchSnapshot()
    })
})