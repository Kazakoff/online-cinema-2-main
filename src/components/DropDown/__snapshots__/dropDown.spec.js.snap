// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown component should render DropDown component 1`] = `
"<div className=\\"btn-group dropdown\\" style={{...}}>
  <button type=\\"button\\" className=\\"btn whiteBtn dropdown-toggle\\" data-toggle=\\"dropdown\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\">
    some label
  </button>
  <div className=\\"dropdown-menu\\" style={{...}}>
    <div style={{...}}>
      <p>
        1
      </p>
      <p>
        2
      </p>
    </div>
  </div>
</div>"
`;

exports[`ErrorMessage component should render ErrorMessage component 1`] = `
"<div className=\\"btn-group dropdown\\" style={{...}}>
  <button type=\\"button\\" className=\\"btn whiteBtn dropdown-toggle\\" data-toggle=\\"dropdown\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\">
    some label
  </button>
  <div className=\\"dropdown-menu\\" style={{...}}>
    <div style={{...}}>
      <p>
        1
      </p>
      <p>
        2
      </p>
    </div>
  </div>
</div>"
`;
